product_harmonized_system_country/models/hs_code_country.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2024 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class HSCodeCountry(models.Model):
+    _name = "hs.code.country"
+    _description = "H.S. Code Country"
+
+    country_ids = fields.Many2many(
+        string="Countries",
+        comodel_name="res.country",
+    )
+    local_code = fields.Char(
+        required=True,
+    )
+    hs_code = fields.Char(
+        string="H.S. Code",
+        compute="_compute_hs_code",
+        store=True,
+        precompute=True,
+    )
+    hs_code_id = fields.Many2one(
+        string="H.S. Code",
+        comodel_name="hs.code",
+        ondelete="cascade",
+    )
+
+    @api.constrains("country_ids")
+    def _check_country_ids(self):
+        for hs_code_country in self:
+            if any(
+                set(hs_code_country.country_ids.ids) & set(c.country_ids.ids)
```

Comment:
humm 'hs_code_country.country_ids.ids' this is not a set? id not u can compare 'hs_code_country.country_ids' whit 'c.country_ids' because are record.set .
but beter 😄

Reply:
I think you can avoid the loop:

other_product_hs_code_countries = hs_code_country.hs_code_id.hs_code_country_ids - hs_code_country
if set(hs_code_country.country_ids.ids) & set(other_product_hs_code_countries.country_ids.ids):
    raise ValidationError(

Reply:
@Tisho99 @ValentinVinagre Yes, that's better. Thank you!